bpf/lbmap/lbmap.go

```diff
@@ -173,13 +208,94 @@ func LookupRevNat(key RevNatKey) (RevNatValue, error) {
 	return revnat.Convert(), nil
 }
 
+func gcd(x, y uint16) uint16 {
+	for y != 0 {
+		x, y = y, x%y
+	}
+	return x
+}
+
+type rrState struct {
+	index int
+	cw    uint16
```

Comment:
Current Weight in computing the sequence and is initialized to zero. This is the state needed across calls to compute the next weight in the sequence.